core/templates/pages/exploration-editor-page/exploration-editor-page.component.spec.ts

```diff
@@ -0,0 +1,677 @@
+        spyOn(window, '$').and.returnValue(element);
+        var animateSpy = spyOn(element, 'animate').and.callThrough();
+        spyOn(angular, 'element').withArgs('#tutorialStateInteraction').and
+          .returnValue(<any>{
+            offset: () => ({
+              top: 20
+            })
+          });
+
+        ctrl.EDITOR_TUTORIAL_OPTIONS[3].fn(true);
+
+        expect(animateSpy).toHaveBeenCalledWith({
+          scrollTop: (20 - 200)
+        }, 1000);
+      });
+
+      it('should change preview tab element scroll top when calling fn' +
+        ' property function on index 5 of EDITOR_TUTORIAL_OPTIONS', function() {
+        var element = angular.element('div');
+        // @ts-ignore is being used in order to ignore JQuery properties that
+        // should be declared.
+        spyOn(window, '$').and.returnValue(element);
+        var animateSpy = spyOn(element, 'animate').and.callThrough();
+        spyOn(angular, 'element').withArgs('#tutorialPreviewTab').and
+          .returnValue(<any>{
+            offset: () => ({
+              top: 5
+            })
+          });
+
+        ctrl.EDITOR_TUTORIAL_OPTIONS[5].fn(true);
+
+        expect(animateSpy).toHaveBeenCalledWith({
+          scrollTop: (5 - 200)
+        }, 1000);
+      });
+
+      it('should change state interaction element scroll top when calling fn' +
+        ' property function on index 5 of EDITOR_TUTORIAL_OPTIONS', function() {
+        var element = angular.element('div');
+        // @ts-ignore is being used in order to ignore JQuery properties that
+        // should be declared.
+        spyOn(window, '$').and.returnValue(element);
+        var animateSpy = spyOn(element, 'animate').and.callThrough();
+        spyOn(angular, 'element').withArgs('#tutorialStateInteraction').and
+          .returnValue(<any>{
+            offset: () => ({
+              top: 20
+            })
+          });
+
+        ctrl.EDITOR_TUTORIAL_OPTIONS[5].fn(false);
+
+        expect(animateSpy).toHaveBeenCalledWith({
+          scrollTop: (20 - 200)
+        }, 1000);
+      });
+
+      it('should change preview tabn element scroll top when calling fn' +
+        ' property function on index 7 of EDITOR_TUTORIAL_OPTIONS', function() {
+        var element = angular.element('div');
+        // @ts-ignore is being used in order to ignore JQuery properties that
+        // should be declared.
+        spyOn(window, '$').and.returnValue(element);
+        var animateSpy = spyOn(element, 'animate').and.callThrough();
+        spyOn(angular, 'element').withArgs('#tutorialPreviewTab').and
+          .returnValue(<any>{
+            offset: () => ({
+              top: 5
+            })
+          });
+
+        ctrl.EDITOR_TUTORIAL_OPTIONS[7].fn(true);
+
+        expect(animateSpy).toHaveBeenCalledWith({
+          scrollTop: (5 - 200)
+        }, 1000);
+      });
+
+      it('should change state interaction element scroll top when calling fn' +
```

Comment:
I see a lot of duplicate tests (very similar except for the index). Can we do this in a loop?

Reply:
They have some divergences between the element which is being manipulated in Jquery and the argument itself can change how Jquery will manipulate the element. However, I'll add some descriptions in this part.